components/rhs_thread/rhs_thread.tsx

```diff
@@ -32,45 +35,60 @@ export function renderThumbHorizontal() {
     return (<div/>);
 }
 
-export function renderThumbVertical(props) {
+export function renderThumbVertical(props: Props) {
     return (
         <div
             {...props}
             className='scrollbar--vertical'
         />);
 }
 
-export default class RhsThread extends React.Component {
-    static propTypes = {
-        posts: PropTypes.arrayOf(PropTypes.object).isRequired,
-        channel: PropTypes.object.isRequired,
-        selected: PropTypes.object.isRequired,
-        previousRhsState: PropTypes.string,
-        currentUserId: PropTypes.string.isRequired,
-        previewCollapsed: PropTypes.string.isRequired,
-        previewEnabled: PropTypes.bool.isRequired,
-        socketConnectionStatus: PropTypes.bool.isRequired,
-        actions: PropTypes.shape({
-            removePost: PropTypes.func.isRequired,
-            selectPostCard: PropTypes.func.isRequired,
-            getPostThread: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Props = {
+    posts: Post[];
+    channel: Partial<Channel> | null;
+    selected: Record<string, any>;
+    previousRhsState?: string;
+    currentUserId: string;
+    previewCollapsed: string;
+    previewEnabled: boolean;
+    socketConnectionStatus: boolean;
+    actions: {
+        removePost: (post: ExtendedPost) => (dispatch: DispatchFunc, getState: GetStateFunc) => void;
+        selectPostCard: (post: Record<string, any>) => Record<string, any>;
+        getPostThread: (rootId: string, root?: boolean) => void;
+    };
+}
+
+type State = {
+    selected?: Record<string, any>;
+    windowWidth?: number;
+    windowHeight?: number;
+    isScrolling: boolean;
+    topRhsPostId: number | string;
+    openTime: number;
+    postsArray?: Record<string, any>[];
+    isBusy?: boolean;
+}
+
+export default class RhsThread extends React.Component<Props, State> {
+    private scrollStopAction: DelayedAction;
+    public refs: any; // todo: check
 
-    static getDerivedStateFromProps(props, state) {
-        let updatedState = {selected: props.selected};
+    public static getDerivedStateFromProps(props: Props, state: State) {
+        let updatedState: Record<string, any> = {selected: props.selected}; // todo: delete later
         if (state.selected && props.selected && state.selected.id !== props.selected.id) {
             updatedState = {...updatedState, openTime: (new Date()).getTime()};
         }
         return updatedState;
     }
 
-    constructor(props) {
+    public constructor(props: Props) {
         super(props);
 
         this.scrollStopAction = new DelayedAction(this.handleScrollStop);
 
         const openTime = (new Date()).getTime();
+        this.refs = React.createRef();
```

Comment:
This is using the old `ref` API, if you'd like to do that you'd have to change the logic.
For now, `refs` can probably just be an empty object if TypeScript is demanding an initialization.

Reply:
Done